WIP: Retire Old Interface

steve-lad opened the pull request:
Use exclusively AMI interface
Delete oldinterface
Delete srvinterface and move aminterface to Sccp_manager.inc
All calls now pass directly to aminterface
Remove all code related to chan-sccp prior v433, as not compatible with this version.

Significantly reduces code base and complexity, to facilitate debug, maintainability, and enhancements

steve-lad committed:
Retire Old Interface

Use exclusively AMI interface
Delete oldinterface
Delete srvinterface and move aminterface to Sccp_manager.inc
All calls now pass directly to aminterface

steve-lad committed:
Add error message for chan-sccp compatibility

Ad error message in install if chan-sccp version is too low
Minor formatting

steve-lad committed:
Bug fix

Missing closing bracket

steve-lad committed:
Remove incompatible files

Simplify compatibility checking. This version is only compatible with chan-sccp 433 or greater.

steve-lad committed:
Return RevisionNum

Return RevisionNum - omitted in previous commit

steve-lad committed:
Clean up install.php

Remove obsolete/non-applicable code

dkgroot commented:
Perfect that was intended but never finished. Thanks

dkgroot commented on Sccp_manager.inc/aminterface.class.php:
```diff
@@ -51,7 +54,14 @@ public function __construct($parent_class = null)
+                        'user' => '',
+                        'pass' => '',
+                        'port' => '5038',
+                        'tsoket' => 'tcp://',
```
tsoket => socket or socket type ?

steve-lad replied:
Russian for socket? or maybe PHP reserved. Will look and change, but is correct

dkgroot commented on Sccp_manager.inc/aminterface.class.php:
```diff
@@ -428,16 +433,16 @@ function core_sccp_reload()
+        //Initialise result array
+        $result = array( 'RevisionHash' => '', 'vCode' => 0, 'RevisionNum' => 0, 'futures' => '', 'Version' => 0);
         if ($this->_connect_state) {
             $_action = new \FreePBX\modules\Sccp_manager\aminterface\SCCPConfigMetaDataAction();
```
So happy that SCCPConfigMetaData is actually being used now :-)

dkgroot replied:
Note: We can use SCCPConfigMetaData to generate some of the array lists. For example some of the drop downs are based on enums, and SCCPConfigMetaData can provide these. We ought to be able to get SCCPConfigMetaData once during installation or every XX days can cache the json content in a temp file, if speed is a problem. Have a look at what SCCPConfigMetaData contains if you are not familiar.

steve-lad replied:
My thoughts exactly - I want to move much of the initialisation to the installer

dkgroot commented on Sccp_manager.inc/aminterface.class.php:
```diff
@@ -447,39 +452,24 @@ function getSCCPVersion()
-                if ($metadata['RevisionNum'] >= 10403) { // new method, RevisionNum is incremental
-                    $result['vCode'] = 432;
-                }
-                if ($metadata['RevisionNum'] >= 10491) { // new method, RevisionNum is incremental
```
I am happy to see we are leaving behind the old versions and concentrate only on last

steve-lad replied:
I think that before I commit, we should clone develop to say legacy, so that if someone cannot use the latest version of chan-sccp, there is a fallback
There are still bugs in PhantoVl even after PR108

dkgroot replied:
You can just set/add a tag that says "legacy" to save this moment in time.
Added https://github.com/chan-sccp/sccp_manager/releases/tag/legacy for you

dkgroot commented on views/server.info.php:
```diff
@@ -26,20 +26,26 @@
 $db_Schema = $this->dbinterface->validate();
 $mysql_info = $this->dbinterface->get_db_sysvalues();
-$compatible = $this->srvinterface->get_compatible_sccp();
+$compatible = $this->aminterface->get_compatible_sccp();
```
We should choose for either camelCase or SnakeCase for functions/methods but not both. I prefer camelCase, but that does not matter.

steve-lad committed:
Update Installer - WIP

Add database definitions in module.xml
Catch exception if PHP Zip not installed

steve-lad committed:
Update README

Add required PHP classes
Delete Russian version as not maintained

steve-lad committed:
Merge branch 'develop' into RetireOldInterface

steve-lad committed:
Use Traits to simplify Sccp_Manager.class

Move helper functions to trait
Move Ajax to trait (WIP - next step to create subclass to simplify initialisation)
Rename Directories to eliminate punctuation
Add new directory for traits

steve-lad committed:
Name Changes

steve-lad committed:
Create BMO Trait

Move BMO code to separate trait

steve-lad committed:
Remove unused methods

steve-lad committed:
Change Function names

steve-lad committed:
Update Sccp_manager.class.php

Remove unused methods
Substitute single line methods
Clean up page presentation (this changes the previous tab order)

steve-lad committed:
Update Sccp_manager.class.php

Move SccpInfo to first tab leaving General as default

steve-lad committed:
Remove duplicate Technical note folder

steve-lad committed:
Update module.xml

Remove duplicate tag

steve-lad committed:
Update Sccp_manager.class.php

Reset settings page tab order

steve-lad committed:
Update Sccp_manager.class.php

Remove unnecessary page code - this version only works with chan-sccp >433

steve-lad committed:
Update Sccp_manager.class.php

Remove unnecessary tests

steve-lad committed:
Update Sccp_manager.class.php

Update version

steve-lad committed:
Update xmlinterface.class.php

Clean up

steve-lad committed:
Modify installer

Change load sequence
Delegate table creation to module.xml
Correct module.xml tables
Correct timezone typo

steve-lad committed:
Minor Formatting

steve-lad committed:
Update page.html.php

steve-lad committed:
Update dbinterface.class.php

Standardise SQL quoting and style

steve-lad committed:
Impliment FreePBX Updater

Add json file to repo

steve-lad committed:
Update module.xml
